### PR TITLE
remove Bio.PDBParser (Issue #777)

### LINKED ISSUE #777: Remove the BioPython PDBParser for topology and trajectory readers
In the discussion in #775 it became clear that the BioPython Parser should be removed completely.

### PR DESCRIPTION
**Pull request by orbeckst**
Fixes #777 

This is a clean-up/merge branch from PR #812 by @jdetle. Below is the original summary:

Changes made in this Pull Request:
- Changed all relevant doc strings to eliminate references to Biopython Readers/Writers and parsers
- Removed the Biopython sloppy structure builder, (entire ~/MDAnalysis/coordinates/pdb directory)
- Replaced PDBReader, PDBWriter, PDBParser with their 'Primitive' counterparts
- Kept 'Primitive' PDBReader, Writer and Parser that inherit from their PDB counterparts and have no format name in the way that the PDBReader has format='PDB', added deprecation warnings indicating this change
- Changed PDB tests to remove known failures of Biopython
- Updated doc strings to remove references to Primitive PDB Readers, Writers and Parsers.
 

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by orbeckst** [comment]
@jdetle I am squashing a few commits to make it look nicer but then it should be good to go.

**Comment by orbeckst** [comment]
My git-foo is not strong enough. I couldn't figure out how to cleanly squash. It is difficult because you used merges as you went along and that intertwines the history of your PR with the develop branch. When I then try an interactive rebase I would have to amend commits that are already in develop's history. Perhaps I could cherry-pick everything onto a clean branch but that's too much effort so I will merge as is.

In the future, make a *separate* branch in your own clone (do not work on your own *develop* and avoid making a PR from jdetle/develop to mdanalysis/develop). It's generally better to occasionally rebase your own branch against develop than to merge because that keeps all your commits in a nice order and it's easy to change them.

**Comment by jdetle** [comment]
Thanks, I'll make sure my future commits are extra clean to make up for this ugliness.

**Comment by orbeckst** [comment]
Maybe it's just difficult for *me* to rebase a merge ;-).

Anyway, I cherry-picked your commits onto a clean branch, resolved the cherry picks, and then rebased the cherry-picked history and pushed it to this branch. Have a look if everything is still here.

I only did minor amends (eg in CHANGELOG).

Btw, don't bother adding descriptions of new tests to the testsuite/CHANGELOG. It really only chronicles changes to the test infrastructure. Tests for everything else are a given so as soon as something shows up in package/CHANGELOG you should assume that it comes with tests.

**Comment by orbeckst** [comment]
Re: why is the rebase difficult (as far as I understand it): Once you attempt the `git rebase -i 744e6444120840d34b2602b06acf6929291c29ef` you will see that there are many instances where your commits and another commit conflict. These other commits come from the merge (use eg `gitk` for a nice visual representation). You should not be amending these commits because they already exist in the published history on *develop*. If you were to amend them, their SHA1 would change (they change identity) and so once you try to merge this branch into develop, you'd either find out that you need to change the existing published commits, too (bad, you should not `git push -f` to published branches that other people use) or you re-introduce the amended commit into the history as an (almost) duplicate, which is also not pretty.

(I hope someone else will explain it better/more correctly...)

**Comment by jdetle** [comment]
Okay, I think I need to draw out some branches to properly understand this. But I think I understand the idea, rewriting history is something we want to avoid when using version control.

**Comment by orbeckst** [comment]
Well, with git we actually like rewriting history *before* we publish it. "Publish" means "going into develop or master". We use history rewriting a lot on feature/fix branches (where `git push -f` is fair game because everyone knows that history can change there... `git pull` is generally quite good about following forced updates.)

**Comment by orbeckst** [comment]
@kain88-de , @jbarnoud or @richardjgowers – do you want to review the PR?

**Comment by jdetle** [comment]
Actually, I'll look it over right now.

**Comment by orbeckst** [comment]
Rebased against develop and force-pushed
```
git rebase develop
git push -f
```
(so that it can be easily merged)

**Comment by kain88-de** [comment]
> I need to draw out some branches

[git-branching](https://pcottle.github.io/learnGitBranching/) is a good tool to get familiar with git branches.

About rebasing merge commits. That is impossible. During normal squashing you know that every commit has just one parent. Then you go down the branch until you find the commit you are looking for and combine the diffs for all the commits you are squashing. For merge commits this isn't really possible, they have two parents. Now if you to down parent A and squash there all the history from parent B would not match up with A, there is no way to decide when B's changes could go into A again.

Squashing all commits between a merge and then cherry-picking those into a new branch is about the best idea. It is possible to remove merges but that is complicated and error prone.

File: testsuite/MDAnalysisTests/test_topology.py [hunk 1]
```diff
@@ -165,16 +165,6 @@ def setUp(self):
     def tearDown(self):
         del self.universe
 
-    def test_correct_parser(self):
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/test_topology.py]: This test should still exist, just without the perm bit, ie
```
-        ret = get_parser_for(self.topology)
-
-        assert_equal(self.parser, ret)
```

Comment (orbeckst) [inline testsuite/MDAnalysisTests/test_topology.py]: Just added it back, rebased and pushed.

**Comment by orbeckst** [comment]
Removed tests asking for "permissive".

**Comment by orbeckst** [comment]
I think this is good to be merged.

**Pull request merged**